pinot-integration-tests/src/test/java/org/apache/pinot/compat/tests/StreamOp.java

```diff
@@ -71,18 +118,170 @@ public void setInputDataFileName(String inputDataFileName) {
     _inputDataFileName = inputDataFileName;
   }
 
+  public String getTableConfigFileName() {
+    return _tableConfigFileName;
+  }
+
+  public void setTableConfigFileName(String tableConfigFileName) {
+    _tableConfigFileName = tableConfigFileName;
+  }
+
+  public String getRecordReaderConfigFileName() {
+    return _recordReaderConfigFileName;
+  }
+
+  public void setRecordReaderConfigFileName(String recordReaderConfigFileName) {
+    _recordReaderConfigFileName = recordReaderConfigFileName;
+  }
+
   @Override
   boolean runOp() {
-    System.out.println("Produce rows into stream " + _streamConfigFileName + " and verify rows in tables "
-        + _tableConfigFileNames);
+    switch(_op) {
+      case CREATE:
+        return createKafkaTopic();
+      case PRODUCE:
+        return produceData();
+    }
     return true;
   }
 
-  public List<String> getTableConfigFileNames() {
-    return _tableConfigFileNames;
+  private boolean createKafkaTopic() {
+    try {
+      Properties streamConfigMap = JsonUtils.fileToObject(new File(_streamConfigFileName), Properties.class);
+      String topicName = streamConfigMap.getProperty("stream.kafka.topic.name");
```

Comment:
We can use `topicName` and `numPartitions` as the keys. We don't need to namespace it in the yaml file